### PR TITLE
Fixed replaceOne model's return type

### PR DESCRIPTION
**Pull request by lpizzinidev**
Addresses #12347

### CODE REVIEW

**Review by AbdelrahmanHafez — CHANGES_REQUESTED** [review]
LGTM, thanks 👍

Can you also add TS tests to prevent future regressions?

**Comment by lpizzinidev** [comment]
@AbdelrahmanHafez Sure 👍

File: test/types/models.test.ts [hunk 1]
```diff
@@ -466,3 +467,17 @@ async function gh12286() {
...
+  }
+  const schema = new Schema<IUser>({
+    name: { type: String, required: true }
+  });
+
+  const User = model<IUser>('User', schema);
+
+  const user = await User.replaceOne({}, {});
+  expectType<UpdateWriteOpResult>(user);
+};
```
Comment (Uzlopak) [inline test/types/models.test.ts:483]:
```suggestion
}

```

**Review by AbdelrahmanHafez — APPROVED** [review]
LGTM, thanks 👍

File: test/types/models.test.ts [hunk 2]
```diff
@@ -466,3 +467,17 @@ async function gh12286() {
...
+    name: string;
+  }
+  const schema = new Schema<IUser>({
+    name: { type: String, required: true }
+  });
+
+  const User = model<IUser>('User', schema);
+
+  const user = await User.replaceOne({}, {});
+  expectType<UpdateWriteOpResult>(user);
```
Comment (AbdelrahmanHafez) [inline test/types/models.test.ts:482]:
```suggestion
  const replaceOneResult = await User.replaceOne({}, {});
  expectType<UpdateWriteOpResult>(replaceOneResult);
```

**Review by hasezoey — APPROVED** [review]
LGTM
The mentioned issue this PR resolves should maybe be added under "development" (or through original message `fixes` or `closes`)

**Pull request merged**